app/services/reports/acquisition_report_service.rb

```diff
@@ -36,15 +36,29 @@ def report
     end
 
     # @return [Integer]
-    def distributed_disposable_diapers
-      @distributed_disposable_diapers ||= organization
+    def distributed_loose_disposable_diapers
+      @distributed_loose_disposable_diapers ||= organization
                                .distributions
                                .for_year(year)
                                .joins(line_items: :item)
                                .merge(Item.disposable)
                                .sum('line_items.quantity')
     end
 
+    def distributed_disposable_diapers_from_kits
+      @distributed_diapers_from_kits ||= organization
+      .distributions
+      .for_year(year)
+      .joins(line_items: { item: :kit })
+      .merge(Item.disposable)
+      .where.not(kits: { id: nil })
+      .sum("line_items.quantity")
+    end
+
```

Comment:
I'm not convinced that this gives us what we need, but I'm not an ActiveRecord expert.  

  @dorner,  will this give the sum of disposable diapers distributed in kits for the year?

Reply:
This is going to be tricky because we have to dip into line items twice for different reasons.   On the one hand we need to know what kits are distributed,  and on the other, we need to know the disposables items that are in the kits.  And we need the.. is aggregate the right word? .. the total of all of those.

Reply:
Yep. I think we're going to need a manual SQL query here - trying to rely solely on Rails will make it extra confusing IMO.

Reply:
I'm quite certain you're right on that.   @jadekstewart3 -- do you happen to be an SQL fiend?   (I don't want us to run roughshod over this if it's in your wheelhouse.)

Reply:
@cielf I am not super great with SQL yet, its something I've ben meaning to brush up on. I can give it a go if you'd like, and ping yall if I hit a wall.

Reply:
Sounds good -- I admit I don't know off the top of my head how to get the whole summing thing to work properly, so I'd have to do some research.

Reply:
Hey @dorner  Do you have a good sense of what is needed here?  If not, I can do some research / consulting with the former DBA I live with and see what I can do.

Reply:
@jadekstewart3 I _think_ this should do what you want. I haven't had time to verify it yet.

```sql
SELECT SUM("line_items"."quantity" * "kit_line_items"."quantity")
FROM "distributions" 
INNER JOIN "line_items" ON "line_items"."itemizable_type" = 'Distribution' AND "line_items"."itemizable_id" = "distributions"."id" 
INNER JOIN "items" ON "items"."id" = "line_items"."item_id" 
INNER JOIN "kits" ON "kits"."id" = "items"."kit_id" 
INNER JOIN "line_items" "kit_line_items" ON "items.id" = "kit_line_items"."id"
INNER JOIN "items" "kit_items" ON "kit_items.id" = "kit_line_items"."item_id"
INNER JOIN "base_items" ON "base_items"."partner_key" = "kit_items"."partner_key" 
WHERE "distributions"."organization_id" = $2 
  AND (extract(year from issued_at) = '$3') 
  AND (lower(base_items.category) LIKE '%diaper%') 
  AND NOT (lower(base_items.category) LIKE '%cloth%' OR lower(base_items.name) LIKE '%cloth%')
```

Just so you know... I did this with Ruby and it looks like there are only 3 organizations who distributed kits with diapers in them in 2023. :(

Reply:
(nods) I didn't expect a lot of kit use for diapers  -- my impression is that kits are a more common distribution strategy for period products, and that's waiting in the wings.

Reply:
I updated the test and the query does seem to work. The test wasn't adding line items in the kit. The kit "item" is just a placeholder, usually called "Kit", just used to attach the kit to line items. (We want to rework this really badly.) You need to add line items to the kit. So the test should look like this.

```ruby
      toddler_disposable_kit = create(:kit, :with_item, organization: organization)
      infant_disposable_kit = create(:kit, :with_item, organization: organization)
      infant_cloth_kit = create(:kit, :with_item, organization: organization)

      within_time = Time.zone.parse("2020-05-31 14:00:00")

      create(:base_item, name: "Toddler Disposable Diaper", partner_key: "toddler diapers", category: "disposable diaper")
      create(:base_item, name: "Infant Disposable Diaper", partner_key: "infant diapers", category: "infant disposable diaper")
      create(:base_item, name: "Infant Cloth Diaper", partner_key: "infant cloth diapers", category: "cloth diaper")

      toddler_disposable_kit_item = create(:item, name: "Toddler Disposable Diapers", partner_key: "toddler diapers")
      infant_disposable_kit_item = create(:item, name: "Infant Disposable Diapers", partner_key: "infant diapers")
      infant_cloth_kit_item = create(:item, name: "Infant Cloth Diapers", partner_key: "infant cloth diapers")

      toddler_disposable_kit.line_items.first.update!(item_id: toddler_disposable_kit_item.id, quantity: 5)
      infant_disposable_kit.line_items.first.update!(item_id: infant_disposable_kit_item.id, quantity: 5)
      infant_cloth_kit.line_items.first.update!(item_id: infant_cloth_kit_item.id, quantity: 5)

      infant_distribution = create(:distribution, organization: organization, issued_at: within_time)
      toddler_distribution = create(:distribution, organization: organization, issued_at: within_time)

      create(:line_item, quantity: 10, item: toddler_disposable_kit.item, itemizable: toddler_distribution)
      create(:line_item, quantity: 10, item: infant_disposable_kit.item, itemizable: infant_distribution)
      create(:line_item, :distribution, quantity: 10, item: infant_cloth_kit.item, itemizable: infant_distribution)

      service = described_class.new(organization: organization, year: within_time.year)

      expect(service.disposable_diapers_from_kits_total).to eq(100)
```

Reply:
The expected number is 100 here because it's the kit quantity in the distribution times the quantity of the eligible items in the kit.